File: 260304_Activity/app/src/main/java/br/gov/sp/cps/activity/MainActivity.java
package br.gov.sp.cps.activity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.widget.Button;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    private Button btnChamaT2;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });



        btnChamaT2 = findViewById(R.id.btnChamaT2);
        btnChamaT2.setOnClickListener(view -> {
            Intent intent = new Intent(this, MainActivity2.class);


            startActivity(intent);
        });

        Log.i("Ciclo de vida", "T1-onCreate");
    }

    @Override
    public void onResume(){
        super.onResume();
        Log.i("Ciclo de vida","T1-onRestart");
    }

    @Override
    public void onStart(){
        super.onStart();
        Log.i("Ciclo de vida", "T1-onStart");
    }

    @Override
    public void onStop(){
        super.onStop();
        Log.i("Ciclo de vida", "T1-onStop");
    }

    @Override
    public void onRestart(){
        super.onRestart();
        Log.i("Ciclo de vida", "T1-onRestart");
    }

    @Override
    public void onDestroy(){
        super.onDestroy();
        Log.i("Ciclo de vida", "T1-onDestroy");
    }

    @Override
    public void onPause(){
        super.onPause();
        Log.i("Ciclo de vida", "T1-onPause");
    }

}

File: AnoMesDia2/app/src/main/java/com/example/ano_mes_dia/MainActivity.kt
package com.example.ano_mes_dia

import android.os.Bundle
import android.view.ViewDebug
import android.widget.ArrayAdapter
import android.widget.Button
import android.widget.EditText
import android.widget.Spinner
import android.widget.TextView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.collection.LruCache
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import java.time.LocalDate

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        val spnTipoConversao = findViewById<Spinner>(R.id.spn_TipoConversao)
        val txtAniversario = findViewById<EditText>(R.id.txt_Aniversario)
        val txtResult = findViewById<TextView>(R.id.txt_Result)
        val btnConverte = findViewById<Button>(R.id.btn_Conversao)

        spnTipoConversao.adapter = ArrayAdapter<String>(this, android.R.layout.simple_spinner_dropdown_item,
            listOf("Converter para Dias", "Converter para Meses", "Converter para Anos"))

        btnConverte.setOnClickListener {

            if(txtAniversario.text.isEmpty()){
                txtAniversario.error = "Digite a sua idade em anos"
            } else {
                val aniversariotxt = txtAniversario.text.toString()
                val aniversario = aniversariotxt.split("/")


                val anoAni = aniversario[2].toInt()
                val mesAni = aniversario[1].toInt()
                val diaAni = aniversario[0].toInt()
                val result: Int

                val anoAtual = LocalDate.now().year
                val mesAtual = LocalDate.now().monthValue
                val diaAtual = LocalDate.now().dayOfMonth
                val diferencaA = if (anoAtual >= anoAni) {
                    anoAtual - anoAni
                } else {
                    anoAni - anoAtual
                }
                val diferencaM = if (mesAtual >= mesAni) {
                    mesAtual - mesAni
                } else {
                    mesAni - mesAtual
                }
                val diferencaD = if (diaAtual >= diaAni) {
                    diaAtual - diaAni
                } else {
                    diaAni - diaAtual
                }

                val tipoConversao = spnTipoConversao.selectedItem as String

                if (tipoConversao == "Converter para Dias") {
                    result = diferencaA * 360 + diferencaM * 30 + diferencaD
                    txtResult.text = "Você tem $result dias de vida!"
                } else if (tipoConversao == "Converter para Meses") {
                    result = diferencaA * 12 + diferencaM + diferencaD / 30
                    txtResult.text = "Você tem $result meses de vida!"

                } else {
                    result = diferencaA + ((diferencaM + diferencaD / 30) / 12)
                    txtResult.text = "Você tem $result anos de vida!"
                }
            }
           }

    }
}